src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -2873,12 +2873,20 @@ public override void Visit(FirstNameNode node)
                         isConstantNamedFormula = formula.IsConstant;
                     }
                 }
-                else if (lookupInfo.Kind == BindKind.Data)
-                {
-                    if (lookupInfo.Data is IExternalCdsDataSource or IExternalTabularDataSource)
-                    {
-                        _txb.SetMutable(node, true);
-                    }
+                else if (lookupInfo.Kind == BindKind.Data)
+                {
+                    if (lookupInfo.Data is IExternalCdsDataSource or IExternalTabularDataSource)
+                    {
+                        _txb.SetMutable(node, true);
+                    }
+                    else if (lookupInfo.Data is IExternalDataSource ds)
+                    {
+                        _txb.SetMutable(node, ds.IsWritable);
+                    }
+                }
+                else if (lookupInfo.Kind == BindKind.ScopeCollection)
+                {
+                    _txb.SetMutable(node, true);
                 }
```

Comment:
These have been added since PFx introduced a new semantic check for mutable args. Some of mutable collection that PA creates were failing the semantic check.